extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/AbstractWidgetRenderer.java

```diff
@@ -188,10 +188,16 @@ protected String getStateAsNumber(Widget w) {
         if (itemName != null) {
             try {
                 Item item = itemUIRegistry.getItem(itemName);
+                State state = item.getState();
                 if (item.getAcceptedDataTypes().contains(PercentType.class)) {
-                    return escapeURLPath(item.getStateAs(PercentType.class).toString());
+                    state = item.getStateAs(PercentType.class);
                 } else {
-                    return escapeURLPath(item.getStateAs(DecimalType.class).toString());
+                    state = item.getStateAs(DecimalType.class);
+                }
+                if (state != null) {
+                    return escapeURLPath(state.toString());
+                } else {
+                    logger.debug("State '{}' of item '{}' is not a number!", item.getState(), itemName);
```

Comment:
No, item.getState() is never null, it might only be UnDefType.NULL. If state==null here, this is only because the getStateAs() returned null, since it wasn't possible to convert the state.